### PR TITLE
[Bugfix][GLM] Invalidate merged KDA convolution weights after refit

### LINKED ISSUE #55087: [Bug]: GLM-5.3 KDA keeps stale merged convolution weights after refit
### Your current environment

vLLM `0.28.1rc1.dev313+g8f8cc414d` on NVIDIA H100, using the GLM-5.3-Flash implementation from #53906 with `load_format="dummy"` and `enforce_eager=True`.

### 🐛 Describe the bug

GLM-5.3 KDA lazily caches the concatenated q/k/v convolution weights in `_merged_conv_weight`. In an online-RL workflow, vLLM warmup builds this cache from dummy weights; a later weight refit updates `q_conv1d`, `k_conv1d`, and `v_conv1d`, but does not invalidate the cached tensor, so inference continues using the dummy-derived convolution weights.

### Before submitting a new issue...

- [x] I searched existing and past issues for `_merged_conv_weight`, KDA reload/refit, and stale derived tensors.

### PR DESCRIPTION
**Pull request by ItsRoy69**
## Purpose

Fixes #55087.

GLM-5.3 KDA lazily merges the q/k/v convolution weights into a single cached tensor for the fused causal-conv path. During online RL or refit workflows, the underlying q/k/v convolution weights can be updated after warmup, but the derived merged tensor may remain stale. Subsequent inference can then continue using old or dummy weights even after the model weights have been refreshed.

This change:

- adds explicit invalidation for the merged KDA convolution-weight cache
- invalidates the KDA cache before GLM-5.3 model weight loading
- covers both the direct model-loading path and the top-level CausalLM loading path
- keeps the merged-weight cache for normal inference when weights are unchanged
- retains version/storage checks as additional protection against direct tensor mutations
- preserves the hot-path behavior by rebuilding only when needed

This PR is a follow-up to #53906, which introduced the GLM-5.3 implementation.

## Root cause

The KDA layer builds a single merged q|k|v convolution weight tensor and caches it in `_merged_conv_weight`. That cache was not always invalidated when the source q/k/v conv weights were replaced. In online-RL refit/update flows, this allowed stale merged weights to persist across reloads and be reused in later inference steps.

## Fix

- add invalidate_merged_conv_weight() / invalidate_kda_conv_weight_caches() methods to manually clear the stale cache
- invoke invalidation before every load_weights() call so refits always rebuild from fresh weights
- add explicit invalidation hooks for the merged KDA cache
- invoke cache invalidation before GLM-5.3 weight loading so the next access rebuilds from fresh weights

## Test Plan

Run the focused regression test on a CUDA-enabled environment:

```bash
pytest tests/models/glm5next/test_kda.py -v
```

Run the relevant GLM-5.3 model tests:

```bash
pytest tests/models/glm5next -v
```

Run static validation:

```bash
python -m py_compile \
  vllm/models/glm5next/nvidia/kda.py \
  vllm/models/glm5next/nvidia/model.py

ruff check \
  vllm/models/glm5next/nvidia/kda.py \
  vllm/models/glm5next/nvidia/model.py
```

```bash
git diff --check
```

## Test Result

Static validation completed successfully:

- Python compilation passed.
- Ruff passed.
- `git diff --check` passed.

The full CUDA/model pytest suite was not run locally because this environment does not have the repo virtualenv, PyTorch, or CUDA runtime available.

CUDA validation is still required on the GLM-5.3 PR validation environment. The focused regression test should verify that:

1. the merged convolution weight is built and cached
2. a refit using `param.data.copy_()` updates a source convolution weight
3. explicit invalidation clears the stale merged tensor
4. the next access rebuilds the merged tensor from the updated weights

## Test Environment

Local environment:

- Windows
- no repository virtualenv available
- no PyTorch installation available
- no CUDA runtime available

## Scope

Changed files:

- `kda.py`
- `model.py`
- `test_kda.py`

No documentation update is required because this is an internal cache invalidation fix.

---

### Checklist
- [x] The purpose of the PR is documented and links issue #55087.
- [x] The root cause and fix are clearly described.
- [x] The test plan includes focused, model-level, and static validation commands.
- [x] Local environment limitations are documented.
- [x] No documentation update is required for this internal implementation fix.

### CODE REVIEW

File: tests/models/glm5next/test_kda.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+
+from types import SimpleNamespace
+
+import torch
+
+from vllm.models.glm5next.nvidia.kda import Glm5NextLinearAttention
```
Comment (JaredforReal) [inline tests/models/glm5next/test_kda.py]: the import path is outdated

**Review by JaredforReal — COMMENTED** [review]
Thanks~ @ItsRoy69 
tested and this pr indeed fixes `Invalidate merged KDA convolution weights after refit`
but I dont really think the diff have achieve the `so-call fixed` below:
>add a version-aware merged-weight cache key in the KDA layer
rebuild the merged tensor when any source conv weight changes

can u update the commits or rewrite the PR description? 
and also some conflict and outdated import path need to be solved

File: tests/models/glm5next/test_kda.py (hunk 1 above)
Comment (ItsRoy69) [reply tests/models/glm5next/test_kda.py]: @JaredforReal made the changes accordingly and rewrote the PR description, and solved the conflict

**Comment by ItsRoy69** [comment]
> Thanks~ @ItsRoy69 tested and this pr indeed fixes `Invalidate merged KDA convolution weights after refit` but I dont really think the diff have achieve the `so-call fixed` below:
> 
> > add a version-aware merged-weight cache key in the KDA layer
> > rebuild the merged tensor when any source conv weight changes
> 
> can u update the commits or rewrite the PR description? and also some conflict and outdated import path need to be solved

@JaredforReal can you recheck ?
